### DISCUSSION TITLE
A few question about using haraka

### DISCUSSION THREAD

**Discussion by WestFarmer** [haraka/Haraka#3232] [Q&A] [answered]
I know a little about mail protocols, please help me to understand.

My goal is to setup private email server

```
haraka: smtp
dovecot: imap + pop3
roundcube: webmail
```

as we are in isolated network, anti-spam and anti-virus is not needed.

questions:

1. How dovecot get emails from haraka, as dovecot plays the` IMAP Server` role, where/how does it find user emails ? 
2. Does haraka support `oauth2 / openid connect` ? something like [A Set of Simple Authentication and Security Layer (SASL) Mechanisms for OAuth](https://datatracker.ietf.org/doc/html/rfc7628)

**Comment by msimerson** [comment, marked as answer]
1. You need to set up a mail spool. There's a variety of ways to do this. The oldest is a unix host and every user on the system has a mail spool file. The MTA delivers to local users and dovecot reads from the users mail file. There's a wide variety of mail virtualization systems out there where users are stored in files, SQL, or directories (LDAP) and the user mechanism points to a Maildir for each user. Configure your MTA and dovecot accordingly.
2. Not yet, someone would need to wrote an auth plugin to add that.
